libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/Classes.hs

```diff
@@ -0,0 +1,707 @@
+
+import Cardano.Ledger.BaseTypes (EpochNo (..), ProtVer (..))
+import Cardano.Ledger.Coin (Coin (..), DeltaCoin (..))
+import qualified Cardano.Ledger.Core as Core
+import Cardano.Ledger.Crypto (Crypto)
+import Cardano.Ledger.Era (Era (..))
+import Cardano.Ledger.Keys (KeyHash, KeyRole (..))
+import Cardano.Ledger.PoolDistr (IndividualPoolStake (..))
+import Cardano.Ledger.Pretty (PDoc)
+import Cardano.Ledger.Shelley.PParams (pvCanFollow)
+import Cardano.Ledger.Shelley.Rewards (Reward (..))
+import Cardano.Ledger.TxIn (TxIn)
+import Cardano.Ledger.UTxO (UTxO (..))
+import Cardano.Ledger.Val (Val (coin, modifyCoin, (<+>)))
+import Data.Default.Class (Default (def))
+import qualified Data.List as List
+import Data.Map.Strict (Map)
+import qualified Data.Map.Strict as Map
+import Data.Maybe.Strict (StrictMaybe (SJust))
+import Data.Set (Set)
+import qualified Data.Set as Set
+import Data.Word (Word64)
+import Lens.Micro
+import Numeric.Natural (Natural)
+import Test.Cardano.Ledger.Alonzo.Serialisation.Generators ()
+import Test.Cardano.Ledger.Babbage.Serialisation.Generators ()
+import Test.Cardano.Ledger.Constrained.Combinators (errorMess, suchThatErr)
+import Test.Cardano.Ledger.Core.Arbitrary ()
+import Test.Cardano.Ledger.Generic.PrettyCore (pcTxOut, pcVal)
+import Test.Cardano.Ledger.Generic.Proof (
+  AllegraEra,
+  GoodCrypto,
+  Proof (..),
+  Reflect (..),
+  ShelleyEra,
+  unReflect,
+ )
+import Test.Cardano.Ledger.Shelley.Generator.Constants (defaultConstants)
+import Test.Cardano.Ledger.Shelley.Generator.Update (genShelleyPParamsUpdate)
+import Test.Cardano.Ledger.Shelley.Serialisation.EraIndepGenerators ()
+import Test.Cardano.Ledger.Shelley.Serialisation.Generators ()
+import Test.Cardano.Ledger.ShelleyMA.Serialisation.Generators ()
+import Test.QuickCheck (
+  Arbitrary (..),
+  Gen,
+  Positive (..),
+  choose,
+  chooseInt,
+  elements,
+  frequency,
+  shuffle,
+  vectorOf,
+ )
+import Prelude hiding (subtract)
+
+import Cardano.Ledger.Shelley.Governance (ShelleyPPUPState (..))
+import qualified Cardano.Ledger.Shelley.Governance as Core (GovernanceState (..))
+import qualified Cardano.Ledger.Shelley.PParams as Core (ProposedPPUpdates (..))
+
+-- import  Cardano.Ledger.Conway.Governance(ConwayTallyState(..))
+-- import Lens.Micro
+
+-- =====================================================================
+-- Partitioning a value into a bunch of pieces, that sum to that value
+
+-- | Generate a list of length 'size' that sums to 'total', where the minimum element is (>= 'smallest')
+intPartition :: [String] -> String -> Int -> Int -> Int -> Gen [Int]
+intPartition msgs typname smallest size total
+  | size == 0 = errorMess (zeroCount "intPartition" total) msgs
+  | size > total =
+      errorMess
+        ( "Can't partition "
+            ++ show total
+            ++ " into "
+            ++ show size
+            ++ " positive pieces at type "
+            ++ typname
+        )
+        msgs
+  | size < 1 =
```

Comment:
I changed ti it   'size <= 0'

Reply:
:smile: But that's the same thing as `< 1`

My comment was about the fact that the `==0` case will never be triggered here because of the clause earlier

Reply:
This has been fixed